client/packages/common/src/authentication/hooks/usePermissionCheck.ts

```diff
@@ -0,0 +1,30 @@
+import { useEffect, useRef } from 'react';
+import { AppRoute } from '@openmsupply-client/config';
+import { UserPermissionNodePermission } from '../../types/schema';
+import { AuthError, useAuthContext } from '../AuthContext';
+import { useLocalStorage } from '../../localStorage';
+import { useNavigate } from 'react-router-dom';
+import { RouteBuilder } from '../../utils/navigation';
+
+export const usePermissionCheck = (
+  permission: UserPermissionNodePermission
+) => {
+  const { userHasPermission } = useAuthContext();
+  const navigate = useNavigate();
+  const [error, setError] = useLocalStorage('/auth/error');
+  const previous = useRef(error);
+
+  useEffect(() => {
+    if (!userHasPermission(permission)) {
+      setError(AuthError.PermissionDenied);
+    }
+  }, []);
+
+  useEffect(() => {
```

Comment:
was using the `addListener` but the component remains mounted, which means that the listener fires on auth timeout etc. Thought to extend the `AuthenticationAlert` and put the navigate in there but decided to use the effect instead